nilearn/datasets/tests/test_neurovault.py

```diff
@@ -712,13 +776,20 @@ def test_fetch_neurovault(tmp_path):
     os.chmod(os.path.join(tmp_path, "neurovault"), stat.S_IREAD | stat.S_IEXEC)
     if os.access(os.path.join(tmp_path, "neurovault"), os.W_OK):
         return
+
     with pytest.warns(UserWarning):
         neurovault.fetch_neurovault(data_dir=tmp_path)
 
 
 def test_fetch_neurovault_errors(request_mocker):
+    """Test that errors are raised when the server returns an error code.
```

Comment:
Maybe more accurate

```suggestion
    """Test that errors are logged when the server returns an error code.
```

Can we instead capture the error with pytest's [capsys](https://docs.pytest.org/en/7.4.x/reference/reference.html#capsys) fixture and match it to a pattern in the error? Something like:
```python
def test_fetch_neurovault_errors(capsys):
    request_mocker.url_mapping["*"] = 500
    data = neurovault.fetch_neurovault()
    captured = capsys.readouterr()
    match = re.search("500 Error", captured.err)
    assert match is not None
    assert len(data.images) == 0
```

Reply:
Uuuh. Nice idea. Me like it.

Reply:
done